<p>
The <code>alt</code> attribute provides a textual alternative to an image.
It is used whenever the actual image cannot be rendered.
Common reasons for that include:
</p>

<ul>
  <li>The image can no longer be found</li>
  <li>Slow internet connections, while waiting for the image to download</li>
  <li>Visually impaired users using a screen reader software</li>
  <li>Images loading is disabled, e.g. to reduce data consumption on mobile phones</li>
</ul>

<h2>Noncompliant Code Example</h2>

<pre>
&lt;img src="foo.png" /&gt;       &lt;!-- Non compliant --&gt;
</pre>

<h2>Compliant Code Example</h2>

<pre>
&lt;img src="foo.png" alt="Information handled by foo.png" /&gt;       &lt;!-- When foo.png handles information --&gt;
&lt;img src="foo.png" alt="" /&gt;                                     &lt;!-- When foo.png is purely decorative --&gt;
</pre>
